railties/lib/rails/application/bootstrap.rb

```diff
@@ -73,7 +73,7 @@ module Bootstrap
 
       initializer :initialize_event_reporter, group: :all do
         Rails.event.raise_on_error = config.consider_all_requests_local
-        Rails.event.debug_mode = Rails.env.development?
+        Rails.event.debug_mode = Rails.env.development? || config.log_level == "debug"
```

Comment:
Note: I initially went for this:

```ruby
Rails.event.debug_mode = config.log_level == "debug"
```

which I thought would be enough. However that was breaking this test:

https://buildkite.com/rails/rails/builds/123977/steps/table?jid=019aab00-bba9-40d3-9a83-3d642e88683b#019aab00-bba9-40d3-9a83-3d642e88683b/1344-1353

That test is unrelated to the structured events, so I wasn't sure if I was supposed to change it.
Let me know 🙇